executor/src/test/java/org/k8loud/executor/kubernetes/DeleteResourceTest.java

```diff
@@ -0,0 +1,113 @@
+package org.k8loud.executor.kubernetes;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.k8loud.executor.exception.KubernetesException;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.k8loud.executor.util.Util.getFullResourceName;
+
+class DeleteResourceTest extends BaseTest {
+    private static final Long GRACE_PERIOD_SECONDS = 5L;
+
+    @ParameterizedTest
+    @MethodSource
+    void testDeletingPod(String namespace, String resourceName, String resourceType, Long gracePeriodSeconds)
+            throws KubernetesException {
+        // given
+        Pod pod = new PodBuilder().withNewMetadata().withName(resourceName).endMetadata()
+                .withNewSpec()
+                .addNewContainer()
+                .withName(resourceName)
+                .withImage("nginx:1.7.9")
+                .addNewPort().withContainerPort(80).endPort()
+                .endContainer()
+                .endSpec()
+                .build();
+        client.pods().inNamespace(namespace).resource(pod).create();
+        assertNotNull(client.pods().inNamespace(namespace).withName(resourceName).get());
+
+        // when
+        String res = kubernetesService.deleteResource(namespace, resourceName, resourceType, gracePeriodSeconds);
+        Pod pod1 = client.pods().inNamespace(namespace).withName(resourceName).get();
+
+        // then
+        assertNull(pod1);
+        assertEquals(String.format("Resource %s deleted", getFullResourceName(resourceType, resourceName)), res);
```

Comment:
Create a String with expected result, don't use methods that are used to create this result